### PR TITLE
resolve self type

### LINKED ISSUE #215: Reference to self is added to index instead of being resolved to class

### LINKED ISSUE #313: @return self is misunderstood
image
image
File: https://github.com/slimphp/Slim/blob/3.x/Slim/Http/Request.php

### PR DESCRIPTION
**Pull request by jens1o**
resolves #313, resolves #215

### CODE REVIEW

**Comment by jens1o** [comment]
I'll add a test for the params now.

**Comment by jens1o** [comment]
It's more complex than I thought. I'm still working on it. I need to look after a location where I can plug me in.

**Comment by jens1o** [comment]
Ok, I have this. And I don't know why it isn't returning what it should... When debugging, it's working and it's returning the right... I'm following the stack trace now, but I'm unsure...

File: tests/Server/TextDocument/HoverTest.php [hunk 1]
```diff
@@ -46,6 +46,20 @@ public function testHoverForSelfReturnTypeDefintion()
...
+    {
+        // class Something
+        // get hover for Something::getInstance() (returns a instance of Something)
+        $reference = $this->getReferenceLocations('Something')[1];
+        $result = $this->textDocument->hover(
+            new TextDocumentIdentifier($reference->uri),
+            $reference->range->end
+        )->wait();
+        $this->assertEquals(new Hover([
+            new MarkedString('php', "<?php\nclass Something")
```
Comment (jens1o) [inline tests/Server/TextDocument/HoverTest.php]: this fails... It doesn't get resolved even though it's been returned...

**Comment by felixfbecker** [comment]
Are you still working on this?

**Comment by jens1o** [comment]
As far as I can see, this was already implemented by @roblourens . (I'm closing)

**Pull request closed without merging**